fingerprint2.js

```diff
@@ -51,7 +51,27 @@
       }
       return target
     },
-    get: function (done) {
+    getOrDefaultPromise: function(defaultResult, defaultComponents){
+      var that = this;
+      return new Promise(function(resolve, reject) {
+        that.getPromise()
+          .then(resolve)
+          .catch(function(e) {
+            resolve([defaultResult, defaultComponents, e])
+          });
+      })
+    },
+    getPromise: function() {
+      var that = this;
+      return new Promise(function(resolve, reject) {
+        // aggregates all resolve function args into one array-like object (resolve() supports only one arg)
+        var argJoiner = function() {
```

Comment:
Please use an explicit, "non-magic" function here. It's ok to "hardcode" the function signature. It helps people understand the code.

Reply:
Will do